crates/ra_hir/src/ty/infer.rs

```diff
@@ -967,8 +988,51 @@ impl<'a, D: HirDatabase> InferenceContext<'a, D> {
                 Ty::unit()
             }
             Expr::For { iterable, body, pat } => {
-                let _iterable_ty = self.infer_expr(*iterable, &Expectation::none());
-                self.infer_pat(*pat, &Ty::Unknown, BindingMode::default());
+                use crate::nameres::PerNs;
+                use crate::resolve::Resolution::Def;
+                use crate::code_model::ModuleDef::Trait;
+                use super::traits::ProjectionPredicate;
+                use super::ProjectionTy;
+
+                let iterable_ty = self.infer_expr(*iterable, &Expectation::none());
+
+                let into_iter_path = Path {
+                    kind: PathKind::Plain,
+                    segments: vec![
+                        PathSegment { name: Name::from_str("std"), args_and_bindings: None },
```

Comment:
This will not work for `#![no_std]` projects.

Reply:
rustc has special casing where it uses `core` if `no_std` is on and `std` otherwise. We don't handle `no_std` at all yet though.